proposals/pr002-hunkspace.md

```diff
@@ -0,0 +1,545 @@
+localized data from sources other than [CLDR][cldr] More detail is in the
+[Detailed Proposal](#h.us1rqruwdp1h) section, after the proposal is motivated
+by a few use cases.
+
+This document proposes a global, shared *hunkspace* (better names always
+welcome) of localized "[hunk][hunk]" data for the ICU4X Data Provider.  It
+doesn't necessarily affect the ICU4X data provider implementation, but does
+affect the data model.   Specifically the proposal is to make the data model
+more general and open to independent third party extensions.
+
+> The terminology proposed in [ICU4X Locale Data Pipeline][icudp] is used.  Any
+> inconsistent word use is a bug, and should be fixed.
+
+[cldr]: http://cldr.unicode.org
+[dp]: https://github.com/unicode-org/icu4x/blob/master/docs/data-pipeline.md
+[ff]: https://github.com/filmil
+[hunk]: https://github.com/unicode-org/icu4x/blob/master/docs/data-pipeline.md
+[icudp]: https://github.com/unicode-org/icu4x/blob/master/docs/data-pipeline.md
+
+# Prior Art
+
+The *ICU4X Locale Data Pipeline* document proposes a [data model][dm] that fits
+the current use case of CLDR data. The data storage method is abstracted from
+the API by using *Data Providers*.  The API is using the
+`key:request/value:response` pattern.  See [Request][rq] in the source code for
+the example key.
+
+[dm]: https://github.com/unicode-org/icu4x/blob/master/docs/data-pipeline.md
+[rq]: https://github.com/unicode-org/icu4x/blob/master/components/data-provider/src/lib.rs
+
+# Use cases
+
+> This section lists a few use cases where a shared hunkspace are helpful.
+
+## Third party language packs
+
+The use case comes from [Fuchsia OS][fx], though it is not a
+unique need. The explanation below gives the context of the use case, and ends
+with the description of the case itself.
+
+Fuchsia software is *ephemeral* (downloaded on demand) and is
+*trusted* (verified prior to execution). So, if a process runs on Fuchsia,
+there is a cryptographic proof that one can use to verify the provenance of the
+software. See the [overview of Fuchsia][fxov] for the reasons behind this
+design choice, as it is out of scope here.
+
+[fx]: https://fuchsia.dev
+[fxov]: https://fuchsia.dev/fuchsia-src/concepts
+
+Fuchsia's software delivery is at odds with the usual approach of
+building and distributing software language packs for large software
+products. Here, a software product is "large" if it satisfies the
+following conditions:
+
+1.  The volume of locale-specific customization is large.
+
+    The volume is estimated with respect to some common human measure;
+    e.g. having tens of thousands of localizable resources is likely a
+    large volume of messages, corresponding to a large volume of
+    pertinent use cases.
+
+2.  The user base speaks 100+ languages.
+
+    This means there will always be languages which will be impractical
+    for the publisher to support. And there will always be languages for
```

Comment:
nit: grammar/unnecessary separation:

Change From:
>  which will be impractical for the publisher to support. And there will always be languages... 

To
> which will be impractical for the publisher to support and languages...

Reply:
I prefer shorter sentences. Even if they come out a bit odd.